### ISSUE TITLE
Changing model architecture when test (inference)

### ISSUE THREAD

**Issue by ComputerMath** [mihaidusmanu/d2-net#82]
Hello

I read the paper and the partition of the code in which the inference works.
As the paper and coda says, the inference model change the avg pooling from stride 2 to 1 and use dilated convolution unlike trained model.

In here, I'm curious about two things.
1) How could these ideas be driven?  I haven't read any paper or seen model in which the architecture of model changes when the model infers unlike training. 

2) Is there any reason Why the model wasn't set to be fixed to test-version model in train time at initial?
Does this different architecture between train - test bring better results?

Thanks in advance :)

**Comment by mihaidusmanu** [comment]
Hey. We were initially looking for ways to obtain better local features using our approach from pre-trained (ImageNet) architectures, which is why we switched to 2x2 average pool w. stride 1 + dilated convolutions. In this way, we keep the same receptive field while increasing the feature map resolution.

Training with dilated convolutions is 4x more memory expensive and the final results in our initial experiments were quite similar so we decided to keep the base architecture for faster and more memory efficient training.
